### ISSUE TITLE
[Feature Request]: Change configuration by server

### ISSUE THREAD

**Issue by sumneko** [microsoft/language-server-protocol#874]
Sometimes the server needs to save some state, for example, ignore the specified warning.
At present, there are two ways to solve this problem:
1. Generate a custom profile in the working directory. However, when a user opens a single file, this method cannot be used.
2. Assuming the type of the editor, modify the configuration of it. This method is very unusual.

So I think LSP should provide a way to modify the editor's configuration file.

**Comment by puremourning** [comment]
I completely disagree. Nothing in the current spec even requires that such a file exists. Not all clients would be able to implement this for the exact same reason servers can’t.

**Comment by mickaelistria** [comment]
This IMO seems server-specific and internal. I don't think it belongs to the spec.
There is already "didChangeConfiguration" that can be used to send configuration from client to server, and also, the LS can just check for changes in such configuration file by itself without the client sending a specific notification. So it also seems like the requested operation doesn't really enable anything that's impossible yet.

**Comment by sumneko** [comment]
The significance of LSP is that the server does not need to care about the specific implementation of the editor. 
In VSCode, the paths of configuration files for the stable, insider and Linux versions are different. Other editors may not support modifying the configuration (by informing the server through `ClientCapabilities`).
I believe that 'workspace/configuration' and 'workspace/didChangeConfiguration' exist for the same reason.

**Comment by angelozerr** [comment]
IMHO I think this feature is a very cool feature.

We have started to implement this feature in https://github.com/redhat-developer/vscode-quarkus to exclude with CodeAction a given property to ignore the validation. CodeAction update the client settings.

Here a demo with the usecase:

![ExcludePropertyDemo](https://user-images.githubusercontent.com/1932211/70125688-9716e200-1677-11ea-99b6-3bbd573f4623.gif)

Today we can read configuration from the client side and know when configuration changed on client side, but why we cannot have the other direction (server which updates client configuration). Our usecase with vscode quarkus is a good sample I think. It avoids to the user to update settings at the hand.

We have implemented with generic mean this feature with a Command in https://github.com/redhat-developer/vscode-quarkus/blob/master/src/lsp-commands.ts but it should be improved (today teh update supports only String array).
